src/nvim/ex_docmd.c

```diff
@@ -9256,6 +9270,19 @@ put_view (
           || put_eol(fd) == FAIL) {
         return FAIL;
       }
+
+      if (fputs("call add(s:buffer_names, [bufname('%'),'", fd) < 0
+          || ses_fname(fd, wp->w_buffer, flagp, false) == FAIL
+          || fputs("'])", fd) < 0
+          || put_eol(fd) == FAIL) {
+        return FAIL;
+      }
+
+      if (fputs("file ", fd) < 0
+          || ses_fname(fd, wp->w_buffer, flagp, false) == FAIL
+          || put_eol(fd) == FAIL) {
+        return FAIL;
```

Comment:
test passes with only this change. Are the other changes actually needed?

Maybe there's some edge case, but I'm going to merge it with only this change for now. Can always revert back to the more complex version if needed.